internal/pkg/agent/application/enroll/options.go

```diff
@@ -177,6 +178,12 @@ func FromFleetConfig(cfg *configuration.FleetAgentConfig) EnrollOptions {
 		options.URL = cfg.Client.Hosts[0]
 	}
 
+	// URL at this point may not be parsable by url.Parse (missing protocol)
+	if host := options.URL; len(host) > 0 && !strings.HasPrefix(host, string(remote.ProtocolHTTPS)+"://") &&
+		!strings.HasPrefix(host, string(remote.ProtocolHTTP)+"://") {
+		options.URL = string(cfg.Client.Protocol) + "://" + host
```

Comment:
If I understand correctly, in case we have a fleet URL that does not start with `http://` or `https://` we prepend whatever is in `cfg.Client.Protocol` + `://` to the URL.
 
 I did a quick snippet on [go playground](https://go.dev/play/p/xe_zGJXr3Rd?v=)  just to see what parsing some URLs would return and it would seem to work fine for no protocol, http or https.
 
 Even if we don't want to use [`url.Parse`](https://pkg.go.dev/net/url#Parse) here, wouldn't it be safer to try to match `://` and see if we get a scheme preceding that?